src/Fantomas.Tests/ModuleTestsWithEmptyLineBeforeNestedModuleBody.fs

```diff
@@ -0,0 +1,361 @@
+﻿module Fantomas.Tests.ModuleTestsWithEmptyLineBeforeNestedModuleBody
+
+open Fantomas
+open NUnit.Framework
+open FsUnit
+open Fantomas.Tests.TestHelper
+
+let config = { config with EmptyLineBeforeNestedModuleBody = true }
+
+[<Test>]
+let ``module abbreviation``() =
+    formatSourceString false "module ES = Microsoft.FSharp.Quotations.ExprShape" config
+    |> should equal """module ES = Microsoft.FSharp.Quotations.ExprShape
+"""
+
+[<Test>]
+let ``module with functions``() =
+    formatSourceString false "module internal MyModule = let x = 42" config
+    |> prepend newline
+    |> should equal """
+module internal MyModule =
+
+    let x = 42
+"""
+
+[<Test>]
+let ``open modules``() =
+    formatSourceString false """
+    // comment1
+    open System.IO
+    // comment2
+    open System""" { config with ReorderOpenDeclaration = true }
+    |> prepend newline
+    |> should equal """
+// comment1
+open System
+// comment2
+open System.IO
+"""
+
+[<Test>]
+let ``sort open modules doesn't mess comments up``() =
+    formatSourceString false """
+module internal Fantomas.CodePrinter
```

Comment:
Please only add tests that have an impact related to the code changes you did.
If the test doesn't involve a nested module it shouldn't really be in here.